core/src/components/accordion/accordion.ts

```diff
@@ -524,7 +562,34 @@ function createAccordionItem(
 				itemVisible$.update((c: boolean) => !c);
 			},
 		},
-		directives: {collapseDirective: bindDirectiveNoArg(itemTransition.directives.directive), accordionItemDirective: noop},
+		directives: {
+			toggleDirective: toggleDirective,
+			buttonDirective: mergeDirectives(
+				toggleDirective,
+				createAttributesDirective(() => ({
+					attributes: {
+						type: 'button',
+						disabled: itemDisabled$,
+						class: itemButtonClass$(),
+					},
+					classNames: {'au-accordion-item-button': true},
+				})),
+			),
+			headerDirective: createAttributesDirective(() => ({attributes: {class: itemHeaderClass$()}, classNames: {'au-accordion-item-header': true}})),
+			bodyDirective: createAttributesDirective(() => ({attributes: {class: itemBodyClass$()}, classNames: {'au-accordion-item-body': true}})),
+			bodyContainerDirective: mergeDirectives(
+				bindDirectiveNoArg(itemTransition.directives.directive),
+				createAttributesDirective(() => ({
+					attributes: {
+						id: computed(() => `${itemId$()}-body-container`),
+						class: itemBodyContainerClass$(),
+						'aria-labelledby': computed(() => `${itemId$()}-toggle`),
+					},
+					classNames: {'au-accordion-item-body-container': true},
+				})),
+			),
+			accordionItemDirective: noop,
```

Comment:
Should the *accordionItemDirective* not be created like it was done for the *starDirective* in the rating ?

Reply:
It's a different approach because part of the directive is created inside the accordion, not in the accordion item. I could have created the attribute directive and then merged it inside the accordion but I think it is clearer to handle both at the same time, in the accordion